### ISSUE TITLE
Postgres does not work with volumes when hosted on ACA

### ISSUE THREAD

**Issue by LERoise** [dotnet/aspire#6671]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Adding a Postgresdb using `WithDataVolume` fails to start when deployed to Azure.

All the resourses are created and the postgres container is deployed to the container apps spins up but goes into the state of pending because it does not have the permissions to the storage account to create the initdb:

**Container log:**
```
chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.
 
The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".
 
Data page checksums are disabled.
 
initdb: error: could not change permissions of directory "/var/lib/postgresql/data": Operation not permitted
fixing permissions on existing directory /var/lib/postgresql/data ... 
```

### Expected Behavior

The container to start and db to be created and persisted.

### Steps To Reproduce

Create a new Aspire host project, add Postgres db and a default web api project:
```
var builder = DistributedApplication.CreateBuilder(args);

var pg = builder.AddPostgres("pgtest")
    .WithDataVolume()
    .AddDatabase("pg-db");

builder.AddProject<Projects.WebApplication1>("webapplication1");

builder.Build().Run();
```

Build and run `azd up`

[Repo](https://github.com/LERoise/Aspire-PG-Test/tree/master)

### .NET Version info

.NET SDK:
 Version:           9.0.100
 Commit:            59db016f11
 Workload version:  9.0.100-manifests.3068a692
 MSBuild version:   17.12.7+5b8665660

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.100\

.NET workloads installed:
 [aspire]
   Installation Source: VS 17.11.35431.28
   Manifest Version:    8.2.2/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.2.2\WorkloadManifest.json
   Install Type:              Msi

Configured to use loose manifests when installing new manifests.

Host:
  Version:      9.0.0
  Architecture: x64
  Commit:       9d5a6a9aa4

.NET SDKs installed:
  8.0.202 [C:\Program Files\dotnet\sdk]
  8.0.404 [C:\Program Files\dotnet\sdk]
  9.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.28 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 9.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

**Comment by davidfowl** [comment]
We still have not documented this problem. Volumes don't work with postgres on ACA. It's an azure storage limitation with how the volume is mounted and incompatibility with some linux containers.

**Comment by scornflake** [comment]
Useful to know - is there any workaround or alternative configuration?

**Comment by davidfowl** [comment]
Use this instead: https://learn.microsoft.com/en-us/dotnet/aspire/database/postgresql-entity-framework-integration?tabs=dotnet-cli#azure-postgresql-hosting-integration

cc @IEvangelist we should move this to a separate azure postgres page.

**Comment by scornflake** [comment]
Thanks - to clarify: the only option right now is to use a postgres flexible server? (i.e: not running as a CA in the CAE?)

**Comment by davidfowl** [comment]
That's right. When you use a volume with postgres there are errors because of the type of volume mount (SMB) and it's incompatibility with postgres.

**Comment by eerhardt** [comment]
Closing as external. This is an ACA limitation that Aspire can't fix.

We also don't recommend running databases in containers in ACA, but instead use a hosted database management system.
